descwl_coadd/coadd.py

```diff
@@ -1,16 +1,196 @@
+
+            # set single WCS
+            crval = geom.SpherePoint(
+                wcs.center.ra/coord.radians,
+                wcs.center.dec/coord.radians,
+                geom.radians,
+            )
+            stack_wcs = makeSkyWcs(
+                crpix=crpix,
+                crval=crval,
+                cdMatrix=cd_matrix,
+            )
+            exp.setWcs(stack_wcs)
+            exps.append(exp)
+
+        self._exps = exps
+
+    def _make_warps(self):
+
+        # Setup coadd/warp psf model
+        input_recorder_config = CoaddInputRecorderConfig()
+
+        input_recorder = CoaddInputRecorderTask(
+            config=input_recorder_config, name="dummy",
+        )
+        coadd_psf_config = CoaddPsfConfig()
+        coadd_psf_config.warpingKernelName = self._interp
+
+        # warp stack images to coadd wcs
+        warp_config = afw_math.Warper.ConfigClass()
+
+        # currently allows up to lanczos5, but a small change would allow
+        # higher order
+        warp_config.warpingKernelName = self._interp
+        warper = afw_math.Warper.fromConfig(warp_config)
+
+        nx, ny = self._coadd_dims
+        sky_box = geom.Box2I(
+            geom.Point2I(0, 0),
+            geom.Point2I(nx-1, ny-1),
+        )
+
+        wexps = []
+        weight_list = []
+        for i, exp in enumerate(self._exps):
+
+            # Compute variance weight
+            stats_ctrl = afw_math.StatisticsControl()
+            stat_obj = afw_math.makeStatistics(
+                exp.variance,
+                exp.mask,
+                afw_math.MEANCLIP,
+                stats_ctrl,
+            )
+
+            mean_var, mean_var_err = stat_obj.getResult(afw_math.MEANCLIP)
+            weight = 1.0 / float(mean_var)
+            weight_list.append(weight)
+
+            wexp = warper.warpExposure(
+                self._coadd_wcs,
+                exp,
+                maxBBox=exp.getBBox(),
+                destBBox=sky_box,
+            )
+
+            # Need coadd psf because psf may not be valid over the whole image
+            ir_warp = input_recorder.makeCoaddTempExpRecorder(i, 1)
+            good_pixel = np.sum(np.isfinite(wexp.image.array))
+            ir_warp.addCalExp(exp, i, good_pixel)
+
+            warp_psf = CoaddPsf(
+                ir_warp.coaddInputs.ccds,
+                self._coadd_wcs,
+                coadd_psf_config.makeControl(),
+            )
+            wexp.getInfo().setCoaddInputs(ir_warp.coaddInputs)
+            wexp.setPsf(warp_psf)
+            wexps.append(wexp)
+
+        self._wexps = wexps
+        self._weights = weight_list
+        self._input_recorder = input_recorder
+        self._coadd_psf_config = coadd_psf_config
+
+    def _make_coadd(self):
+        input_recorder = self._input_recorder
+
+        # combine stack images using mean
+        stats_flags = afw_math.stringToStatisticsProperty("MEAN")
+        stats_ctrl = afw_math.StatisticsControl()
```

Comment:
Does the stack know to convert to surface brightness?

Reply:
I have no idea